services/webnn/ort/graph_builder_ort.cc

```diff
@@ -240,15 +243,26 @@ std::string GraphBuilderOrt::GetOperandName(uint64_t operand_id) {
   }
 }
 
-std::string GraphBuilderOrt::CreateInitializerAsRawData(
-    base::span<const uint32_t> shape,
-    base::span<const uint8_t> data,
-    OperandDataType data_type) {
+template <typename T>
+std::string GraphBuilderOrt::CreateInitializer(base::span<const uint32_t> shape,
+                                               base::span<const T> data,
+                                               OperandDataType data_type,
+                                               bool used_for_shape_inference) {
   std::string name = GetInsertedOperandName(next_operand_id_);
   OperandInfo operand_info{name, data_type, shape};
+  size_t byte_size = sizeof(T) * data.size();
 
-  model_builder_.AddInitializerAsRawData(name, operand_info.int64_shape, data,
-                                         operand_info.onnx_data_type);
+  if (byte_size < kMinExternalDataSize || used_for_shape_inference) {
+    model_builder_.AddInitializerAsRawData(
```

Comment:
The signatures of `AddInitializerAsRawData` and `AddInitializerAsExternalData` seem to be the same. Can we combine them into one method (e.g. `AddInitializer`) and encapsulate the `byte_size` check into it?